buildbot/buildbot/templates/job.html

```diff
@@ -38,15 +38,7 @@ <h2>Job {{ job.name }}</h2>
 </p>
 
 <p>
-    See <a href="{{ url_for('job_log', name=job.name) }}">the full log</a>
-    ({{ job.log | length }} messages).
+    See <a href="{{ url_for('job_log', name=job.name) }}">the full log</a>.
 </p>
-<ul class="log">
-{% for time, text in job.log[-4:] %}
-    <li>
-        <time>{{ time | dt }}</time><br>
-        <pre>{{ text }}</pre>
-    </li>
-{% endfor %}
-</ul>
+<pre class="log">{{ log }}</pre>
```

Comment:
Is this just asking Flask to send the `log.txt` here? Could we instead use some JavaScript to continuously pull in the log and snip it to `LOG_PREVIEW_LINES` so we have a nice streaming log?

Reply:
That string variable `log` is the trimmed version of the text file (which the server has already read and trimmed).

That fancy JavaScript version would be cool! But somewhat hard because it would need to use polling (or something complicated like inotify) because we don't know when a command will write more output to the file. And Server-Sent Events in the JavaScript to do the streaming. And async streaming on the server side, which is an awkward fit for Flask.

Reply:
I meant something even dumber like:
```
setInterval(updateLog, 10*1000)
```
where `updateLog` repeatedly sends a request to the server for `log.txt`. We can have some JavaScript to clear the timer when the state = done | failed.

Reply:
Ah, I see. Yeah, we can try that too (but not in this PR, IMO—that’s still quite a change because it involves multiple JavaScripty requests instead of just this simple page formatting).

Reply:
Sounds good. This is ready to merge in that case.